src/io_uring.c

```diff
@@ -0,0 +1,81 @@
+#include "io_uring.h"
+
+#ifdef HAVE_LIBURING
+#include <liburing.h>
+#include <string.h>
+
+#include "serverassert.h"
+#include "zmalloc.h"
+
+#define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
+
+static size_t io_uring_queue_len = 0;
```

Comment:
The separation between this file and networking.c looks very good now. The only thing I noticed now is the coupling between the io_uring instance stored in `server.io_uring` and this static variable in this file. They are separated but they belong together.

Is it possible to get the queue length from the io_uring, then we don't need to store this variable? From chat conversation we think it's maybe possible to use `io_uring_sq_ready` for this.

If not possible, I think we should consider a new struct containing `struct io_uring` and this counter together in one allocation.

```C
typedef struct {
    struct io_uring ring;
    size_t queue_len;
} io_uring_wrapper; /* Find a better name? */
```